di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java

```diff
@@ -209,43 +212,37 @@ private void persistData(QueryParamsMap queryParamsMap, AuthorizationCode author
     }
 
     private ValidationResult validateQueryParams(QueryParamsMap queryParams) {
-        String redirectUriValue = queryParams.value(RequestParamConstants.REDIRECT_URI);
-        if (Validator.isNullBlankOrEmpty(redirectUriValue)) {
-            return new ValidationResult(false, new ErrorObject(
-                    ERROR_CODE_INVALID_REDIRECT_URI,
-                    "redirect_uri param must be provided",
-                    HttpServletResponse.SC_BAD_REQUEST));
-        }
-
         String responseTypeValue = queryParams.value(RequestParamConstants.RESPONSE_TYPE);
         if (Validator.isNullBlankOrEmpty(responseTypeValue) || !responseTypeValue.equals(ResponseType.Value.CODE.getValue())) {
             return new ValidationResult(false, OAuth2Error.UNSUPPORTED_RESPONSE_TYPE);
         }
 
         String clientIdValue = queryParams.value(RequestParamConstants.CLIENT_ID);
-        if (Validator.isNullBlankOrEmpty(clientIdValue)) {
+        if (Validator.isNullBlankOrEmpty(clientIdValue) || CredentialIssuerConfig.getClientConfig(clientIdValue) == null) {
             return new ValidationResult(false, OAuth2Error.INVALID_CLIENT);
         }
 
+        if (Validator.redirectUrlIsInvalid(queryParams)) {
```

Comment:
It could be worth adding an additional check above this to make sure the redirect_uri has been provided. And that can return its own error of something along the lines of a "Missing redirect uri param" error.

And then this check can specifically be just checking that the provided redirect_uri is valid or not according to our config

Reply:
I think technically not providing a redirect_uri is valid, so I didn't want to enforce it here. I've got a ticket to look at if we can support it. If a client only has one redirect_url registered, they should be able to skip providing one as it's redundant.
Having said that, this validation doesn't actually handle that situation.
I'll add the check now, and then address that situation if we get to that ticket.